util/fmtutil/format_test.go

```diff
@@ -231,3 +233,50 @@ func TestMetricTextToWriteRequestErrorParsingMetricType(t *testing.T) {
 	_, err := MetricTextToWriteRequest(input, labels)
 	require.Equal(t, "text format parsing error in line 3: unknown metric type \"info\"", err.Error())
 }
+
+func TestMakeTimeseries_HistogramInfBucket(t *testing.T) {
+	tests := map[string]*dto.Histogram{
+		"Histogram missing +Inf bucket": {
+			Bucket: []*dto.Bucket{
+				{CumulativeCount: p[uint64](5), UpperBound: p(1.0)},
+				{CumulativeCount: p[uint64](10), UpperBound: p(5.0)},
+			},
+			SampleCount: p[uint64](15),
+		},
+		"Histogram already has +Inf bucket": {
+			Bucket: []*dto.Bucket{
+				{CumulativeCount: p[uint64](5), UpperBound: p(1.0)},
+				{CumulativeCount: p[uint64](10), UpperBound: p(5.0)},
+				{CumulativeCount: p[uint64](15), UpperBound: p(math.Inf(1))},
+			},
+			SampleCount: p[uint64](15),
+		},
+	}
+
+	for name, histogram := range tests {
+		t.Run(name, func(t *testing.T) {
+			wr := &prompb.WriteRequest{}
+			labels := map[string]string{"__name__": "test_histogram"}
+			metric := &dto.Metric{
+				Histogram:   histogram,
+				TimestampMs: p(time.Now().UnixMilli()),
+			}
+
+			require.NoError(t, makeTimeseries(wr, labels, metric))
+
+			var hasInf bool
```

Comment:
it would be more descriptive to just define the expected output in the test case and compare directly, let's not calculate here

Reply:
I can't because `makeTimeseries` attaches the current time timestamps. I could use mock clocks, or come up with other ways to `require.Equal(...)` the output with expected output, but if the purpose of this test is just to make sure the `+Inf` value is present, the this is the fewest lines of code I can come up with to do that.

Thoughts?